src/sentry/search/django/backend.py

```diff
@@ -250,6 +256,28 @@ def query(self, project, tags=None, environment=None, sort_by='date', limit=100,
         else:
             retention_window_start = None
 
+        now = timezone.now()
+        end = parameters.get('date_to') or (now + timedelta(minutes=1))
+        # TODO: Presumably we want to search back to the project's full retention,
+        #       which may be higher than 90 days in the future, but apparently
+        #       `retention_window_start` can be None?
+        start = (parameters.get('date_from')
+                 or retention_window_start
+                 or (now - timedelta(days=90)))
+
+        # This is a punt because the SnubaSearchBackend (a subclass) shares so much that it
+        # seemed better to handle all the shared initialization and then handoff to the
+        # actual backend.
+        return self._query(project, start, end, retention_window_start, group_queryset, tags,
+                           environment, sort_by, limit, cursor, count_hits, paginator_options,
+                           **parameters)
```

Comment:
Not in love with this but I tried tearing it apart into smaller pieces that made more sense and kept hitting barriers. It's lame but at least it's clear? IDK.